source/brailleDisplayDrivers/brailleNote.py

```diff
@@ -190,22 +235,33 @@ def _onReceive(self, command):
 		if not arg:
 			log.debugWarning("Timeout reading argument for command 0x%X" % command)
 			return
-		self._dispatch(command, ord(arg))
+		# #5993: Read the buffer once more if a BrailleNote QT says it's got characters in its pipeline.
+		if command == QT_MOD_TAG:
+			key = self._serial.read(2)[-1]
+			arg2 = _qtKeys.get(ord(key), key)
+		else:
+			arg2 = None
+		self._dispatch(command, ord(arg), ord(arg2) if arg2 is not None else None)
 
-	def _dispatch(self, command, arg):
+	def _dispatch(self, command, arg, arg2=None):
 		space = False
-		if command == THUNB_KEYS_TAG :
+		if command == THUMB_KEYS_TAG:
 			gesture = InputGesture(keys=arg)
+		elif command == SCROLL_WHEEL_TAG:
+			gesture = InputGesture(wheel=arg)
 		elif command == CURSOR_KEY_TAG:
 			gesture = InputGesture(routing=arg)
 		elif command in (DOTS_TAG, DOTS_SPACE_TAG, DOTS_ENTER_TAG, DOTS_BACKSPACE_TAG):
 			if command != DOTS_TAG:
 				space = True
 			if command == DOTS_ENTER_TAG:
-				# Stuppid bug in the implementation
+				# Stupid bug in the implementation
 				# Force dot8 here, although it should be already there
 				arg |= DOT_8
 			gesture = InputGesture(dots=arg, space=space)
+		elif command == QT_MOD_TAG:
+			# BrailleNote QT
+			gesture = InputGesture(qtMod=arg, qtData=arg2)

```

Comment:
Have you actually considered making this a keyboardHandler.KeyboardInputGesture?